### ISSUE TITLE
Feature request: API

### ISSUE THREAD

**Issue by SlimeDog** [OliverSchlueter/FancyHologramsPlugin#3]
Please implement a public API so [pHD](https://www.spigotmc.org/resources/77631/) can consider adding support.

**Comment by OliverSchlueter** [comment]
To get a hologram by name:
```java
Hologram holo = FancyHolograms.getInstance().getHologramManager().getHologram("name");
```

To get a list of all holograms:
```java
Collection<Hologram> holograms = FancyHolograms.getInstance().getHologramManager().getAllHolograms();
```

To show/spawn to a player
```java
hologram.spawn(serverPlayer); // requires a ServerPlayer
HologramSpigotAdapter.fromHologram(hologram).spawn(player); // requires a normal Bukkit Player
```

To hide/remove to a player
```java
hologram.remove(serverPlayer);
HologramSpigotAdapter.fromHologram(hologram).remove(player); // requires a normal Bukkit Player
```

There is no information stored which players see which holograms, but I'll make a HologramSpawnEvent and a HologramRemoveEvent, then you can in theory easily track that.

**Comment by SlimeDog** [comment]
> There is no information stored which players see which holograms, but I'll make a HologramSpawnEvent and a HologramRemoveEvent, then you can in theory easily track that.

`pHD` controls visibility for a player, based on the [factors](https://github.com/SlimeDog/pHD/wiki/How-It-Works).

**Comment by OliverSchlueter** [comment]
@SlimeDog do you need any more information or can this be closed?

**Comment by SlimeDog** [comment]
Has FancyHolograms API covered all the listed items?

**Comment by SlimeDog** [comment]
We're also waiting for more hologram features, like heads and items and blocks, but those are in addition to the `pHD` control features.

**Comment by keehl254** [comment]
Would it be possible to adjust the build settings to include the API as a compile, or to combine the API project and the plugin project together?

The developer gradle repo is only allowing me to see the classes inside of the plugin branch; I can do `FancyHologramsPlugin.get().getHologramsManager()`
But I cannot see any of the events, the Hologram class, nor the HologramData class. As a result, I really can't use the API if I am not just manually adding the API package into my project.

![image](https://github.com/FancyMcPlugins/FancyHolograms/assets/6056735/78c3b645-9c54-4538-a111-283c4d705d85)

The above image should have shown me a list of possible methods and fields inside the Hologram class, but as it cannot be found, it is blank.

**Comment by OliverSchlueter** [comment]
> Would it be possible to adjust the build settings to include the API as a compile, or to combine the API project and the plugin project together?
> 
> The developer gradle repo is only allowing me to see the classes inside of the plugin branch; I can do `FancyHologramsPlugin.get().getHologramsManager()` But I cannot see any of the events, the Hologram class, nor the HologramData class. As a result, I really can't use the API if I am not just manually adding the API package into my project.
> 
> ![image](https://user-images.githubusercontent.com/6056735/257827587-78c3b645-9c54-4538-a111-283c4d705d85.png)
> 
> The above image should have shown me a list of possible methods and fields inside the Hologram class, but as it cannot be found, it is blank.

Yes I'll see what I can do (I'm not the best with maven repos and gradle), thank you for telling

**Comment by OliverSchlueter** [comment]
> We're also waiting for more hologram features, like heads and items and blocks, but those are in addition to the `pHD` control features.

I just released the feature to have item & block holograms (https://github.com/FancyMcPlugins/FancyHolograms/releases/tag/v2.0.3)

Do you need anything else?

**Comment by SlimeDog** [comment]
> Do you need anything else?

I will test it over the next week or so. Thanks.

**Comment by SlimeDog** [comment]
> Do you need anything else?

Command(s) to convert from HolographicDisplays and DecentHolograms would be a big plus.

**Comment by SlimeDog** [comment]
Further testing. Essentials:
- Ability to control when a hologram is visible. If not direct control of visibility -- per user -- then permissions.
- An API to control aspects of individual holograms, including visibility options.

**Comment by SlimeDog** [comment]
A hologram should be a collection of (one or more) related display entities of various types. For me, a typical hologram consists of a block in the middle (vertically) of several text lines.

**Comment by SlimeDog** [comment]
I would like to add FancyHolograms support to [pHD](https://www.spigotmc.org/resources/77631/). It would be helpful if the issues noted above
- combined holographic elements in a single hologram
- bouncing/rotating block elements
- centered location specification

could be addressed. Thanks for your consideration.

**Comment by SlimeDog** [comment]
> You can center the hologram on block with the moveTo command (x.5 y z.5)?

Yep. Works great. Thanks.

> I'll think about bouncing & rotating item/block holograms. That could be pretty cool.

That would be great.

> Combined hologram elements in one single holograms is not planned for now. You would need to position multiple holograms by yourself.

OK. Will do.
